cfme/utils/appliance/db.py

```diff
@@ -263,6 +263,43 @@ def loosen_pgssl(self, with_ssl=False):
         result = client.run_command("systemctl restart {scl}-postgresql".format(scl=scl))
         return result.rc
 
+    def create_db_lvm(self):
+        """
+        Set up a partition for the CFME DB to run on.
+
+        As a work-around for having to provide a separate disk to a CFME appliance
+        for the database, we instead partition the single disk we have and run
+        the DB on the new partition.
+
+        This requires that the appliance's disk has more space than CFME requires.
+        For example, on RHOS, downstream CFME uses 43GB on a disk but the flavor used
+        to deploy the template has a 75GB disk. Therefore, we have extra space which
+        we can partition.
+
+        Note that this is not the 'ideal' way of doing things and should
+        be a stop-gap measure until we are capabale of attaching additional disks to
+        an appliance via automation on all infra types.
+        """
+        self.logger.info("Creating new LVM for db using remaining space on /dev/vda...")
+
+        fstab_line = '/dev/mapper/dbvg-dblv $APPLIANCE_PG_MOUNT_POINT xfs defaults 0 0'
+        commands_to_run = [
+            'parted /dev/vda --script mkpart primary 43GB 48.5GB',
+            'pvcreate /dev/vda3',
+            'vgcreate dbvg /dev/vda3',
+            'lvcreate --yes -n dblv --size 5G dbvg',
+            'mkfs.xfs /dev/dbvg/dblv',
+            'echo -e "{}" >> /etc/fstab'.format(fstab_line),
+            'mount -a'
+        ]
+
+        with self.ssh_client as client:
+            for command in commands_to_run:
+                result = client.run_command(command)
+                if result.failed:
```

Comment:
REQUIRED: If one command failed I don't think it makes sense to run the rest of commands.
And I believe log level should be error not warning

Reply:
Done